controller_list:
  - joint_state_controller
  - b_torobo_arm_controller

# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 500

# Position Controllers ---------------------------------------
b_torobo_arm_controller:
  type: "position_controllers/JointTrajectoryController"
  action_ns: follow_joint_trajectory  # for moveit
  action_type: FollowJointTrajectory  # for moveit (==>type)
  action_default: true                # for moveit (==>default)
  joints:
    - b_torobo_arm/joint_1
    - b_torobo_arm/joint_2
    - b_torobo_arm/joint_3
    - b_torobo_arm/joint_4
    - b_torobo_arm/joint_5
    - b_torobo_arm/joint_6
    - b_torobo_arm/joint_7
  allow_partial_joints_goal: true
  constraints:
    goal_time: 0.5
    stopped_velocity_tolerance: 0.05
    b_torobo_arm/joint_1: {trajectory: 0.01745, goal: 0.01745}
    b_torobo_arm/joint_2: {trajectory: 0.01745, goal: 0.01745}
    b_torobo_arm/joint_3: {trajectory: 0.01745, goal: 0.01745}
    b_torobo_arm/joint_4: {trajectory: 0.01745, goal: 0.01745}
    b_torobo_arm/joint_5: {trajectory: 0.01745, goal: 0.01745}
    b_torobo_arm/joint_6: {trajectory: 0.01745, goal: 0.01745}
    b_torobo_arm/joint_7: {trajectory: 0.01745, goal: 0.01745}
  stop_trajectory_duration: 0.5
  state_publish_rate: 20
  action_monitor_rate: 20
